---
output: beamer_presentation
header-includes: \usepackage{graphicx}
---


```{r setup, include = F}

rm(list=ls())

#Load knitr 
library(knitr)

#Set some options
options(na.action='na.fail')
opts_knit$set(root.dir = '..')
opts_chunk$set(echo=FALSE,
               warning=FALSE,
               message=FALSE,
               fig.width=5,
               fig.height=3,
               fig.path = "../Figures/")

```

```{r load and clean, include = F}

#Load and clean data
source('Scripts/GWAS_sims_LOAD.R')
source('Scripts/GWAS_sims_FUNCTIONS.R')
source('Scripts/GWAS_sims_CLEAN.R')

```



## Genome-wide association studies in natural populations: managing expectations and avoiding error

Susan E. Johnston & Lewis G. Spurgin

\vspace{0.3cm}

\begin{center}
\includegraphics[width=0.8\textwidth]{Images/manhattan.png}
\end{center}

\vspace{0.3cm}

\begin{center}
\includegraphics[width=0.9\textwidth]{Images/logos.png}
\end{center}






##**G**enome **W**ide **A**ssociation **S**tudy (GWAS)

Experimental design used to detect associations between genetic variants and traits in population samples







----

\begin{center}
\includegraphics[width=1\textwidth]{Images/gwas_success.png}
\end{center}

\begin{flushright}
\tiny{Visscher et al. 2017, AJHG}
\end{flushright}





----

\begin{center}
\includegraphics[width=1\textwidth]{Images/traits.png}
\end{center}






#Key factors to consider in GWAS






##Key factors to consider in GWAS
- Have I missed any variants associated with my trait? (**False negatives**) 
- Have I identified any spurious associations with my trait? (**False positives**)





##Key factors to consider in GWAS

**Factors affecting false positive and false negative rates**

- Architecture of the trait
- Number of individuals sampled
- Number of genetic markers sampled
- Extent of linkage disequilibrium between sampled markers and causal variants






##Key factors to consider in GWAS

**Factors affecting false positive and false negative rates**

- Architecture of the trait
- Number of individuals sampled
- Number of genetic markers sampled
- **Extent of linkage disequilibrium between sampled markers and causal variants**






##Linkage disequilibrium (LD) and GWAS

\begin{center}
\includegraphics[width=1\textwidth]{Images/ld.png}
\end{center}

\vspace{0.4cm}

\begin{flushright}
\tiny{Bush and Moore 2012, PLoS Comput. Biol.}
\end{flushright}




##Linkage disequilibrium (LD) and GWAS

\begin{center}
\includegraphics[width=0.6\textwidth]{Images/indirect_association.png}
\end{center}

\vspace{0.4cm}

\begin{flushright}
\tiny{Bush and Moore 2012, PLoS Comput. Biol.}
\end{flushright}






##Linkage disequilibrium (LD) and GWAS

Present day patterns of LD in a population are the product of:

1. Historical evolutionary processes (e.g. recombination rate, demography, selection etc. - **Ancestral LD**)
2. Nature of the population sample (e.g. family structure, substructure, sample size - **Sample LD**)

**Both ancestral LD and sample LD will vary widely across studies of natural populations**




##Linkage disequilibrium (LD) and GWAS

\begin{center}
\includegraphics[width=1\textwidth]{Images/ld_birds.png}
\end{center}

\vspace{0.3cm}

\begin{flushright}
\tiny{Bosse et al. 2017, Science; Knief et al. 2017, Mol. Ecol.}
\end{flushright}





##GWAS in natural populations

- The power GWAS will vary enormously across natural populations and species with different evolutionary histories
- Sample size (number of individuals/markers) will therefore need to be adjusted accordingly
- This should be considered **before** undertaking sampling and  genotyping





#GWAS in natural populations: a simulation approach



##Study aims

Develop a simulation and analysis pipeline to:

- Illustrate some of the key issues affecting the power and promise of GWAS in natural populations
- Allow other researchers to estimate power in their system before undertaking costly fieldwork/genotyping





##Simulation approach

- We simulated quantitative traits under three **scenarios**, representing diverse natural systems with different population histories
- We **did not** perform exhaustive simulations to establish whih parameters most affect GWAS

\begin{center}
\includegraphics[width=0.8\textwidth]{Images/scenarios.png}
\end{center}





##Simulation approach

Used the software **QMSim** (Sargolzaei and Schenkel 2009, Bioinformatics), to:

1. Simulate historical population to establish **ancestral LD**
2. Simulate recent pedigrees to generate **sample LD**



##Simulation approach

Sample size:

- Five sample sizes tested (250,500,1000,2500,5000)
- Five marker panel sizes tested (5k,20k,50k,100k,200k)
- 100 replicates per simulation



##Simulation approach

Trait architecture:

- 10 QTL on 10 chromosomes (1 QTL per chromosome)
- Gamma distribution of QTL effect sizes
- Heritability of 0.4
- No selection
- Kept constant in all simulations



##Downstream analyses

Used GCTA to:

- Identify significant GWAS hits
- Estimate QTL effect sizes
- Estimate trait heritability (should be 0.4 in all simulations)



#Results

##Scenario 1

- Large, outbred population (historical $N_e$ = 20k)
- No family structure (i.e. no pedigree simulations)

\vspace{0.3cm}

\begin{center}
\includegraphics[width=0.3\textwidth]{Images/herring.png}
\end{center}



##Scenario 1

```{r s1_gwas_qtl_empty}

s1_gwas_qtl <- subset(g1, (SNP.Name %in% paste0("Q", c(1:10)))) %>%
  ddply(
    .(Nids, Nsnps, HistN, Full.Iteration.ID),
    summarise,
    N_Sig_QTL = sum(Sig.Corrected == "yes")
  ) %>%
  ggplot(aes(
    x = factor(Nids),
    y = N_Sig_QTL,
    fill = factor(Nsnps)
  )) +
  ylim(c(0, 12)) +
  geom_hline(aes(yintercept = 10), lty = 2) +
  theme_bw()+
  geom_boxplot()+
  ylab("Number of significant QTL")+
    xlab("Number individuals sampled")+
    labs(fill='Number SNPs sampled')+
  scale_fill_brewer(palette = "Blues")

s1_gwas_qtl

```

``