---
output: beamer_presentation
header-includes: \usepackage{graphicx}
---


```{r setup, include = F}

rm(list=ls())

#Load knitr 
library(knitr)

#Set some options
options(na.action='na.fail')
opts_knit$set(root.dir = '..')
opts_chunk$set(echo=FALSE,
               warning=FALSE,
               message=FALSE,
               fig.width=5,
               fig.height=3,
               fig.path = "../Figures/")

```

```{r load and clean, include = F}

#Load and clean data
source('Scripts/GWAS_sims_LOAD.R')
source('Scripts/GWAS_sims_FUNCTIONS.R')
source('Scripts/GWAS_sims_CLEAN.R')

```



## Genome-wide association studies in natural populations: managing expectations and avoiding error

Lewis G. Spurgin & Susan E. Johnston

\vspace{0.3cm}

\begin{center}
\includegraphics[width=0.8\textwidth]{Images/manhattan.png}
\end{center}

\vspace{0.3cm}

\begin{center}
\includegraphics[width=0.9\textwidth]{Images/logos.png}
\end{center}






----

\begin{center}
\includegraphics[width=1\textwidth]{Images/traits.png}
\end{center}




##Linking genotype and phenotype in the wild

- How heritable is my trait?
- How many loci are associated with my trait?
- What are their relative effect sizes?
- What evolutionary processes are driving their variation?



##**G**enome **W**ide **A**ssociation **S**tudy (GWAS)

Experimental design used to detect associations between genetic variants and traits in population samples



----

\begin{center}
\includegraphics[width=0.8\textwidth]{Images/poll.png}
\end{center}






----

\begin{center}
\includegraphics[width=1\textwidth]{Images/gwas_success_wild.png}
\end{center}

\begin{flushright}
\tiny{Barson et al. 2015 Nature, Cadieu et al. 2009 Science}
\end{flushright}




##Linking genotype and phenotype in the wild

\begin{center}
\includegraphics[width=0.8\textwidth]{Images/boromir.png}
\end{center}



##Linking genotype and phenotype in the wild

- Have I got the power to detect true associations?
- Am I likely to suffer from false positive associations?


##Study aims

Develop a simulation and analysis pipeline to:

- Illustrate some of the key issues affecting the power and promise of GWAS in natural populations
- Allow other researchers to estimate power in their system before undertaking costly fieldwork/genotyping






##Key factors to consider in GWAS

- Architecture of the trait
- Extent of linkage disequilibrium (LD) in the study population
- Number of individuals sampled
- Number of markers sampled
- Analysis used


##Linkage disequilibrium (LD) and GWAS

- Causal mutations probably not typed in GWAS
- We rely on indirect associations with typed SNPs in LD

\begin{center}
\includegraphics[width=1\textwidth]{Images/indirect_association.png}
\end{center}

\vspace{0.3cm}

\begin{flushright}
\tiny{Hirschorn and Daly 2005, Nat. Rev. Gen.}
\end{flushright}



##Linkage disequilibrium (LD) and GWAS

\begin{center}
\includegraphics[width=1\textwidth]{Images/ld_birds.png}
\end{center}

\vspace{0.3cm}

\begin{flushright}
\tiny{Bosse et al. 2017, Science; Knief et al. 2017, Mol. Ecol.}
\end{flushright}





##Study predictions

In simulated populations with different evolutionary histories and family structures:

- Accuracy of heritability estimates and GWAS will improve with increasing sample size (individuals and markers)
- But benefits of increased sampling will vary across scenarios



##Simulation approach

\begin{center}
\includegraphics[width=1\textwidth]{Images/simulation_flow.png}
\end{center}




##Simulation approach

Historical populations and pedigree simulations carried out using QMSim

- Three scenarios, representing diverse natural systems with different population histories
  - **Scenario 1**: Large population ($N_e$ = 25,0000), no family structure
  - **Scenario 2**: Large population ($N_e$ = 25,0000), family structure
  - **Scenario 3**: Small population ($N_e$ = 500), family structure

\begin{center}
\includegraphics[width=0.8\textwidth]{Images/scenarios.png}
\end{center}


##Simulation approach

Trait architecture:

- 10 QTL on 10 chromosomes (1 QTL per chromosome)
- Gamma distribution of QTL effect sizes
- Heritability of 0.4
- No selection
- Kept constant in all simulations


##Simulation approach

- 3 Scenarios x 5 sample sizes x 5 marker panel sizes x 100 replicates = **7500 datasets**
- Estimated heritability and performed GWAS on all datasets using GCTA
- Whole pipeline will be available to download (probably as an R package)




#Results



##Heritability


```{r h2,fig.height = 7,fig.width = 7,fig.align = "center"}

h4_2 <- h4[,colnames(h4) %in% colnames(h1)]
h2_2 <- h2[,colnames(h2) %in% colnames(h1)]


temp <- rbind(h1,h2_2,h4_2)
temp$Scenario <- paste("Scenario",rep(c(1,2,3),c(nrow(h1),nrow(h2_2),nrow(h4_2))))
temp$Scenario2 <- ifelse(temp$Scenario == "Scenario 1","Large pop",
                         ifelse(temp$Scenario == "Scenario 2","Large pop, FS",
                                "Small pop, FS"))
temp$Nids2 <- paste("N =",temp$Nids)
temp$Nids2 <- factor(temp$Nids2,levels = paste("N =",unique(temp$Nids)[order(unique(temp$Nids))]))

temp$Nsnps2 <- paste(temp$Nsnps,"SNPs")
temp$Nsnps2 <- factor(temp$Nsnps2,levels = paste(unique(temp$Nsnps)[order(unique(temp$Nsnps))],"SNPs"))

Fig2 <-  
  ggplot(temp,aes(x = Nids2, y = H2, col = Nids2)) +
  geom_jitter(aes(shape = NA)) +
  geom_hline(yintercept = 0.4,lty = 2)+
  facet_grid(Nsnps2 ~ Scenario2) +
  theme_bw() +
  theme(legend.position = "none",
  axis.text.x = element_text(angle = 90)) +
  ylab("Number of GWAS hits") +
  xlab("")

Fig2

```





##Heritability

```{r h2_2,fig.height = 7,fig.width = 7,fig.align = "center"}

h4_2 <- h4[,colnames(h4) %in% colnames(h1)]
h2_2 <- h2[,colnames(h2) %in% colnames(h1)]


temp <- rbind(h1,h2_2,h4_2)
temp$Scenario <- paste("Scenario",rep(c(1,2,3),c(nrow(h1),nrow(h2_2),nrow(h4_2))))
temp$Scenario2 <- ifelse(temp$Scenario == "Scenario 1","Large pop",
                         ifelse(temp$Scenario == "Scenario 2","Large pop, FS",
                                "Small pop, FS"))
temp$Nids2 <- paste("N =",temp$Nids)
temp$Nids2 <- factor(temp$Nids2,levels = paste("N =",unique(temp$Nids)[order(unique(temp$Nids))]))

temp$Nsnps2 <- paste(temp$Nsnps,"SNPs")
temp$Nsnps2 <- factor(temp$Nsnps2,levels = paste(unique(temp$Nsnps)[order(unique(temp$Nsnps))],"SNPs"))

Fig2 <-  
  ggplot(temp,aes(x = Nids2, y = H2, col = Nids2)) +
  geom_jitter(height = 0) +
  geom_hline(yintercept = 0.4,lty = 2)+
  facet_grid(Nsnps2 ~ Scenario2) +
  theme_bw() +
  theme(legend.position = "none",
  axis.text.x = element_text(angle = 90)) +
  ylab("Number of GWAS hits") +
  xlab("")

Fig2

```




##Number of significant GWAS hits


```{r gwas_nonqtl,fig.width = 7,fig.height = 7,fig.align = "center"}
g4_2 <- g4[,colnames(g4) %in% colnames(g1)]
g2_2 <- g2[,colnames(g2) %in% colnames(g1)]


temp <- rbind(g1,g2_2,g4_2)
temp$Scenario <- paste("Scenario",rep(c(1,2,3),c(nrow(g1),nrow(g2_2),nrow(g4_2))))
temp$Scenario2 <- ifelse(temp$Scenario == "Scenario 1","Large pop",
                         ifelse(temp$Scenario == "Scenario 2","Large pop, FS",
                                "Small pop, FS"))
temp$Nids2 <- paste("N =",temp$Nids)
temp$Nids2 <- factor(temp$Nids2,levels = paste("N =",unique(temp$Nids)[order(unique(temp$Nids))]))

temp$Nsnps2 <- paste(temp$Nsnps,"SNPs")
temp$Nsnps2 <- factor(temp$Nsnps2,levels = paste(unique(temp$Nsnps)[order(unique(temp$Nsnps))],"SNPs"))

Fig2 <-  subset(temp, !(SNP.Name %in% paste0("Q", c(1:10)))) %>%
  ddply(
  .(Nids2, Nsnps2, HistN, Full.Iteration.ID, Scenario2),
  summarise,
  N_Sig_QTL = sum(Sig.Corrected == "yes")
  ) %>%
  ggplot(aes(x = Nids2, y = N_Sig_QTL, col = Nids2)) +
  geom_jitter(aes(shape = NA)) +
    geom_hline(yintercept = 10,lty = 2)+
  facet_grid(Nsnps2 ~ Scenario2) +
  theme_bw() +
  theme(legend.position = "none",
  axis.text.x = element_text(angle = 90)) +
  ylab("Number of GWAS hits") +
  xlab("")

Fig2

```

##Number of significant GWAS hits


```{r gwas_nonqtl_2,fig.width = 7,fig.height = 7,fig.align = "center"}
g4_2 <- g4[,colnames(g4) %in% colnames(g1)]
g2_2 <- g2[,colnames(g2) %in% colnames(g1)]


temp <- rbind(g1,g2_2,g4_2)
temp$Scenario <- paste("Scenario",rep(c(1,2,3),c(nrow(g1),nrow(g2_2),nrow(g4_2))))
temp$Scenario2 <- ifelse(temp$Scenario == "Scenario 1","Large pop",
                         ifelse(temp$Scenario == "Scenario 2","Large pop, FS",
                                "Small pop, FS"))

temp$Nids2 <- paste("N =",temp$Nids)
temp$Nids2 <- factor(temp$Nids2,levels = paste("N =",unique(temp$Nids)[order(unique(temp$Nids))]))

temp$Nsnps2 <- paste(temp$Nsnps,"SNPs")
temp$Nsnps2 <- factor(temp$Nsnps2,levels = paste(unique(temp$Nsnps)[order(unique(temp$Nsnps))],"SNPs"))

Fig2 <-  subset(temp, !(SNP.Name %in% paste0("Q", c(1:10)))) %>%
  ddply(
  .(Nids2, Nsnps2, HistN, Full.Iteration.ID, Scenario2),
  summarise,
  N_Sig_QTL = sum(Sig.Corrected == "yes")
  ) %>%
  ggplot(aes(x = Nids2, y = N_Sig_QTL, col = Nids2)) +
  geom_jitter(height = 0.2) +
    geom_hline(yintercept = 10,lty = 2)+
  facet_grid(Nsnps2 ~ Scenario2) +
  theme_bw() +
  theme(legend.position = "none",
  axis.text.x = element_text(angle = 90)) +
  ylab("Number of GWAS hits") +
  xlab("")

Fig2


```



##Number of significant GWAS peaks

```{r peaks, fig.width = 7,fig.height = 7,fig.align = "center"}
temp <- fp

temp$QTL.Count <- temp$True.QTL.Count + temp$False.QTL.Count

temp$Scenario2 <- ifelse(temp$Scenario == 1,"Large pop",
                         ifelse(temp$Scenario == 2,"Large pop, FS",
                                "Small pop, FS"))
temp$Nids2 <- paste("N =",temp$Nids)
temp$Nids2 <- factor(temp$Nids2,levels = paste("N =",unique(temp$Nids)[order(unique(temp$Nids))]))

temp$Nsnps2 <- paste(temp$Nsnps,"SNPs")
temp$Nsnps2 <- factor(temp$Nsnps2,levels = paste(unique(temp$Nsnps)[order(unique(temp$Nsnps))],"SNPs"))



Fig2 <-  
  ggplot(temp, aes(x = Nids2, y = QTL.Count, col = Nids2)) +
  geom_jitter(height = 0.2) +
      geom_hline(yintercept = 10,lty = 2)+
  ylim(c(0,12)) +
  facet_grid(Nsnps2 ~ Scenario2) +
  theme_bw() +
  theme(legend.position = "none",
  axis.text.x = element_text(angle = 90)) +
  ylab("Number of GWAS hits") +
  xlab("")

Fig2

```


##Number of false positive peaks

```{r false positives, fig.width = 7,fig.height = 7,fig.align = "center"}
temp <- fp


temp$Scenario2 <- ifelse(temp$Scenario == 1,"Large pop",
                         ifelse(temp$Scenario == 2,"Large pop, FS",
                                "Small pop, FS"))
temp$Nids2 <- paste("N =",temp$Nids)
temp$Nids2 <- factor(temp$Nids2,levels = paste("N =",unique(temp$Nids)[order(unique(temp$Nids))]))

temp$Nsnps2 <- paste(temp$Nsnps,"SNPs")
temp$Nsnps2 <- factor(temp$Nsnps2,levels = paste(unique(temp$Nsnps)[order(unique(temp$Nsnps))],"SNPs"))



Fig2 <-  
  ggplot(temp, aes(x = Nids2, y = False.QTL.Count, col = Nids2)) +
  geom_jitter(height = 0.1) +
  facet_grid(Nsnps2 ~ Scenario2) +
  theme_bw() +
  theme(legend.position = "none",
  axis.text.x = element_text(angle = 90)) +
  ylab("Number of GWAS hits") +
  xlab("")

Fig2

```


##Overall conclusions

- Power and promise of GWAS depends on sample size, but also on evolutionary history
- This should be explored before undertaking sampling and analysis, not after
- We have developed a flexible framework for simulating and analysing quantitative trait data
- Come and talk to us if you want to know more!

#Thankyou!

#Thankyou!

#Thankyou!

#Thankyou!


##test

```{r}
plot(rnorm(1000))
```
